packages/vite/src/utils/options-utils.ts

```diff
@@ -29,7 +29,7 @@ export async function getBuildAndSharedConfig(
   const projectRoot = context.workspace.projects[context.projectName].root;
 
   return mergeConfig({}, {
-    mode: options.mode ?? context.configurationName,
```

Comment:
That's actually needed. We want to allow the ability to pick up the configuration name. Is there a reason you are removing this?

Reply:
This assignment creates undesired side effects. When using custom configuration names, they're passed as `mode` down to vite, And vite doesn't know how to resolve them. Here's an example of a target I'm using for a qwik app
```json
   "build-ssr": {
      "defaultConfiguration": "preview",
      "options": {
        "outputPath": "dist/apps/host"
      },
      "configurations": {
        "preview": {
          "ssr": "src/entry.preview.tsx"
        },
        "cloudflare-pages": {
          "ssr": "entry.cloudflare-pages.tsx"
        }
      },
      "executor": "@nrwl/vite:build",
      "dependsOn": ["build"]
    },
```

Obviously `preview` or `cloudflare-pages` are not valid options for `mode`

Reply:
I think we should include `mode` into the options for each configuration instead of using the `configurationName`.

Reply:
I'm linking [this PR](https://github.com/nrwl/nx/pull/13655) here for reference.

Let's discuss this a bit before removing it.

What do you think would be the ideal way? Just rely on `mode` being passed down? We do, however, want to make sure that when someone does `nx build my-app --configuration=production` this is taken into account. Same as when someone's using a build target like `my-app:build:production`. We want that `production` to be taken into account.

If we remove that, then only `mode` will be taken into account, and that's not something we want in this case... :/ 

See the [explanation in the PR I linked](https://github.com/nrwl/nx/pull/13655#issue-1479073443). 

What would be the best way to go around this, in your opinion?

Reply:
I can merge the `ssr` change to unblock you, in the meantime, though!

Reply:
Thanks for the explanation and merging it @mandarini ! 

I actually agree with Jason's point above that it would be better to explicitly specify the `mode` option in the target configuration, because as was pointed above `configuration` and `mode` are not always the same thing. 

I can create another PR with corresponding changes or file a new issue to continue the discussion, so please let me know 🙂

Reply:
For the sake of simplicity we may also try to resolve the `mode` based on `configuration` in case it's a valid one. And if not either omit it or set to `development` for example 🤔

Reply:
Sounds good to me, I think. What do you think @FrozenPandaz ?

Reply:
I want to think about this a bit more if that's ok with you @dmitry-stepanenko . You can set the mode in the configuration, now, as is. You can also set the mode in `vite.config.ts`. If you set the mode in `configuration` I see how it will be overwritten if you pass `--configuration=production` for example. Let me come up with some scenarios, and we can work this out, ok?

Can you once again explain to me your issue at the moment?

Reply:
https://github.com/nrwl/nx/pull/13856

Reply:
Created a PR already! :)

Reply:
Thank you!